assets/prototype/application/utilities/money/useMoneyDisplay.ts

```diff
@@ -0,0 +1,29 @@
+import formatAmount from './formatAmount';
+import useConfig from '../../services/config/useConfig';
+import { CurrencyProps } from '../../valueObjects/config/types';
+
+export type MoneyDisplay = {
+	// the currency sign if the currency displays it before the amount (or '')
+	afterAmount: string;
+	// the currency sign if the currency displays it before the amount (or '')
+	beforeAmount: string;
+	// the full currency config object
+	currency: CurrencyProps;
+	// function for formatting the amount using the correct number of decimal places for the currency
+	formatAmount: (amount: number | string) => string;
+};
+
+const useMoneyDisplay = (): MoneyDisplay => {
+	const { config } = useConfig();
+	const afterAmount = config.currency.signB4 ? '' : config.currency.sign;
+	const beforeAmount = config.currency.signB4 ? config.currency.sign : '';
+	const formatMoney = formatAmount(config.currency.decimalPlaces);
```

Comment:
I like how this function ended up using `decimalPlaces` from config.
And the fact that it is curried it's double win, since it is acting as micro-factory.
So much noise will be moved away from the components that will use this hook.
Amazing job :bow_and_arrow:

Reply:
thank you...
although I'm not sure if I'm totally happy with this yet.

I'm thinking that `formatAmount` isn't entirely clear as to what formatting is occuring, ie:

- is this only controlling the number of decimal places displayed?
- is it adding thousands separators?
- is it adding currency signs?
- is it adding the currency name/abbreviation?

So I may refactor this to add functions for applying all of those other formatting capabilities and/or a single formatter with some kind of options and/or config parameters for controlling the formatting. But that said, it would still ideally keep all of that noise away from the components.